samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -621,20 +596,47 @@ def _enrich_mapped_resources(
         "IMAGE_LAMBDA_FUNCTION": _enrich_image_lambda_function,
     }
 
-    for sam_metadata_resource in sam_metadata_resources:
-        resource_type = sam_metadata_resource.resource.get("values", {}).get("triggers", {}).get("resource_type")
-        sam_metadata_resource_address = sam_metadata_resource.resource.get("address")
-        enrichment_function = resources_types_enrichment_functions.get(resource_type)
-        if enrichment_function is None:
-            raise InvalidSamMetadataPropertiesException(
-                f"The resource type {resource_type} found in the sam metadata resource {sam_metadata_resource_address} "
-                f"is not a correct resource type. The resource type should be one of these values "
-                f"{resources_types_enrichment_functions.keys()}"
+    # create output directory if it doesn't exist
+    if not os.path.exists(output_directory_path):
```

Comment:
can we push the make file creation to be after the enrichment logic. As in case if we find any issue during the enrichment, and the prepare hook failed, we should not create the Makefile, or the metadata file.

Reply:
done